### ISSUE TITLE
Unsafe variable use in certGen.sh

### ISSUE THREAD

**Issue by malcolmdimeglio** [Azure/azure-iot-sdk-c#1859]
tools/CACertificates/certGen.sh appears to be unstable depending on the value of certain variables.

if you use `root_ca_password=""`  - empty string (line 27), then, when you generate your certificate, the script will hang.
Problems:
- the script becomes unresponsive, event to Ctrl+C and Ctrl+D. Only solution is to close the window altogether
- the `generate_device_certificate` uses `${root_ca_password}` **without** double quotes. That means that when the line is parsed, if the password is an empty string, the argument is not counted and there is a shift in the argument parsing. Therefore, `generate_leaf_certificate` will extract the value given as `${openssl_root_config_file}` and save it in `local ca_password`

Other untested problem:
It seems that this problem could also be created in other functions like `generate_root_ca` where `-out ${root_ca_dir}/private/${root_ca_prefix}.key.pem` (line 68). <- If the root_ca_dir happens to have a white space in it, it might cause a problem.

Bottom line is, the use of those variables should be double quoted all throughout the script.

Moreover, there is no verification if hard codded files and paths exist before using them.

**Comment by ericwol-msft** [comment]
This issue is resolved.
